### ISSUE TITLE
Support for otherwise data-less enums with a catchall variant

### ISSUE THREAD

**Issue by jplatte** [Peternator7/strum#116]
Hi! I've got lots of enums that are in the following form of will be converted to it:

```rust
enum Foo {
    Bar,
    Baz,
    #[doc(hidden)]
    _Custom(String),
}
```

The idea being that this enum can represent any string, but some set of known values have dedicated enum variants. Adding variants should be backwards-compatible, which is why `_Custom` is not part of the public interface (otherwise you could rely on the fact that `Foo::from("qux")` returns `Foo::_Custom(String::new("qux"))` rather than a dedicated variant I might want to later introduce). I'm currently assessing whether I need to create my own macro for this or could improve strum to serve my needs (this is also why I opened a few other issues, but those seem like they would definitely be useful to strum users even if I build my own crate). So:

Are you open to a PR that extends strum with some form of attribute that would allow me to return the custrom string in `AsRefStr`? Something like:

```rust
#[derive(strum::AsRefStr)]
enum Foo {
    Bar,
    Baz,
    #[strum(as_ref_str = 0)]
    _Qux(String),
    #[strum(as_ref_str = member)]
    _Quux { member: Cow<'static, str> }
}
```

Further, are you open to a PR that adds a derive macro for `From<String>`, only for enums with exactly one variant containing a `String` and further, data-less variants (or something a bit more flexible that would also cover this use case)?

**Comment by jplatte** [comment]
@Peternator7 ping

**Comment by Peternator7** [comment]
Does the [`default`](https://github.com/Peternator7/strum/wiki/Additional-Attributes) attribute do what you're describing?

**Comment by Peternator7** [comment]
Perhaps with the addition of #86 as well

**Comment by jplatte** [comment]
Yes, if `default`s behavior was changed to only work on tuple variants with a single element of type `String` (or something similar) and use that field for the various derive macros rather than the variant name, this would address my use case. I can open a PR if you want to change `default` in this way – I'd like to also change it from `strum(default = "true")` to `strum(default)`, which would have the added benefit of not being a silent behaviour change for existing code that upgrades to the next release of strum.

Alternatively, this could be implemented as a new attribute, e.g. `strum(other)`.
